icicle/src/poseidon2/kernels.cu

```diff
@@ -0,0 +1,233 @@
+#include "poseidon/poseidon.cuh"
+#include "gpu-utils/modifiers.cuh"
+
+namespace poseidon2 {
+  template <typename S>
+  DEVICE_INLINE S sbox_el(S element, const int alpha)
+  {
+    S result2 = S::sqr(element);
+    switch (alpha) {
+    case 3:
+      return result2 * element;
+    case 5:
+      return S::sqr(result2) * element;
+    case 7:
+      return S::sqr(result2) * result2 * element;
+    case 11:
+      S result8 = S::sqr(S::sqr(result2));
+      return result8 * result2 * element;
+    }
```

Comment:
maybe add assert(false) in default case so the GPU at least throws some error instead of silently do nothing. I assume you check it earlier but still it wouldn't hurt to assert if here too.